Xamarin.Forms.Core/Slider.cs

```diff
@@ -28,16 +28,18 @@ public class Slider : View, IElementConfiguration<Slider>
 			return value;
 		});
 
-		public static readonly BindableProperty ValueProperty = BindableProperty.Create("Value", typeof(double), typeof(Slider), 0d, BindingMode.TwoWay, coerceValue: (bindable, value) =>
+		public static readonly BindableProperty ValueProperty = BindableProperty.Create("Value", typeof(double), typeof(Slider), 0d, BindingMode.TwoWay, (bindable, value) =>
+		{
+			var slider = (Slider)bindable;
+			return (double)value >= slider.Minimum && (double)value <= slider.Maximum;
+		}, coerceValue: (bindable, value) =>
 		{
 			var slider = (Slider)bindable;
 			return ((double)value).Clamp(slider.Minimum, slider.Maximum);
 		}, propertyChanged: (bindable, oldValue, newValue) =>
```

Comment:
I would think coerce is the way to go and it should just re-coerce the value after the min/max change which is how it works in WPF (which of course means you need to add a **CoerceValue** helper method on BindableObject and call it from the change callback of the related properties). Otherwise you have to know the current and new values for all the properties and set them in the correct order which if you're creating a new control from code behind is fine but it's not fine if you're doing this in XAML or binding the properties. And actually it doesn't make sense that the Minimum and Maximum of the Slider currently have both the ValidateValueCallback and the CoerceValueCallback. The ValidateValueCallback should be removed from those as well.